# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1302573049 14400
#      Mon Apr 11 21:50:49 2011 -0400
# Branch stable
# Node ID b4bd0bef55b63d2a60232ef13b9448871d8fd0ad
# Parent  c3d4542a4f062935c8ef1766824b0b3e4f0fe352
getappdata.m: If appdata propery does not exist, return an empty matrix.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2011-04-11  Ben Abbott  <bpabbott@mac.com>
+
+	* miscellaneous/getappdata.m: If appdata propery does not exist, return
+	an empty matrix.
+
 2011-04-08  Rik  <octave@nomad.inbox5.com>
 
 	* linear-algebra/cond.m, linear-algebra/expm.m, linear-algebra/logm.m,
 	linear-algebra/null.m, linear-algebra/orth.m, linear-algebra/rank.m,
 	linear-algebra/rref.m: Improve docstrings.
 
 2011-04-06  Rik  <octave@nomad.inbox5.com>
 
diff --git a/scripts/miscellaneous/getappdata.m b/scripts/miscellaneous/getappdata.m
--- a/scripts/miscellaneous/getappdata.m
+++ b/scripts/miscellaneous/getappdata.m
@@ -24,20 +24,26 @@
 ## Created: 2010-07-15
 
 function val = getappdata (h, name)
 
   if (! (all (ishandle (h)) && ischar (name)))
     error ("getappdata: invalid input");
   endif
 
-  appdata(numel(h)) = struct();
+  ## FIXME - Is there a better way to handle non-existent appdata
+  ## and missing fields?
+  val = cell (numel (h), 1);
+  appdata = struct();
   for nh = 1:numel(h)
-    appdata(nh) = get (h(nh), "__appdata__");
+    try
+      appdata = get (h(nh), "__appdata__");
+    catch
+      appdata.(name) = [];
+    end_try_catch
+    val(nh) = {appdata.(name)};
   end
-  if (nh > 1)
-    val = {appdata.(name)};
-  else
-    val = appdata.(name);
+  if (nh == 1)
+    val = val{1};
   endif
 
 endfunction
 
